### ISSUE TITLE
Invalid react/no-direct-mutation-state in constructor

### ISSUE THREAD

**Issue by emmanuelgautier** [yannickcr/eslint-plugin-react#1382]
In the following special case, the `react/no-direct-mutation-state` rule throw an invalid error

```jsx
class OneComponent extends Component {
  contructor() {
    super();

    class AnotherComponent extends Component {
      constructor() {
        super();
      }
    }

    this.state = {};
  }
}
```

A better way is to not declare a component into another component constructor but that is an invalid linter error .

**Comment by ljharb** [comment]
What error is thrown?

**Comment by emmanuelgautier** [comment]
Error thrown : `Do not mutate state directly. Use setState() react/no-direct-mutation-state`

**Comment by ljharb** [comment]
This seems like a bug - although I can't conceive of why you'd want to create a component in the constructor like that.

**Comment by emmanuelgautier** [comment]
I'm agree with you, this code is not mine. It should be a closure problem to have access into the `AnotherComponent` to some variables declared in the first constructor

**Comment by manolisal** [comment]
I know I am late to the party, but you have made a typo at the constructor of OneComponent class. Changing _contructor_ to _constructor_ removes the eslint warning. 

In your case eslint treats _contructor_ as it would with any other method of OneComponent's class and it warns you that you should not mutate the state this way and use the setState method.
